coretex/cli/commands/task.py

```diff
@@ -0,0 +1,162 @@
+#     Copyright (C) 2023  Coretex LLC
+
+#     This file is part of Coretex.ai
+
+#     This program is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU Affero General Public License as
+#     published by the Free Software Foundation, either version 3 of the
+#     License, or (at your option) any later version.
+
+#     This program is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU Affero General Public License for more details.
+
+#     You should have received a copy of the GNU Affero General Public License
+#     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from typing import Optional, List, Any
+from threading import Thread
+from pathlib import Path
+
+import sys
+import logging
+import subprocess
+
+import click
+
+from ..resources import LOCAL_EXEC_PATH
+from ..modules.user import initializeUserSession
+from ..modules.utils import onBeforeCommandExecute
+from ..modules.project_utils import isProjectSelected
+from ..._task import TaskRunWorker, LoggerUploadWorker
+from ..._task.local.task_config import readTaskConfig
+from ...configuration import loadConfig
+from ...entities import TaskRun, TaskRunStatus, Project, LogSeverity, Log
+
+
+class RunException(Exception):
+    pass
+
+
+def _handleOutput(process: subprocess.Popen, worker: LoggerUploadWorker) -> None:
+    stdout = process.stdout
+    if stdout is None:
+        raise ValueError("stdout is None for subprocess")
+
+    while process.poll() is None:
+        line: str = stdout.readline().decode("UTF-8")
+        if line.strip() == "":
+            continue
+
+        worker.add(Log.create(line.rstrip(), LogSeverity.info))
+        logging.info(line.rstrip())
+
+
+def _handleError(process: subprocess.Popen, worker: LoggerUploadWorker, isEnabled: bool) -> None:
+    stderr = process.stderr
+    if stderr is None:
+        raise ValueError("stderr is None for subprocess")
+
+    lines: List[str] = []
+    while process.poll() is None:
+        line: str = stderr.readline().decode("UTF-8")
+        if line.strip() == "":
+            continue
+
+        worker.add(Log.create(line.rstrip(), LogSeverity.error))
+        lines.append(line.rstrip())
+
+    # Dump stderr output at the end to perserve stdout order
+    for line in lines:
+        if isEnabled and process.returncode == 0:
+            logging.warning(line)
+        elif isEnabled:
+            logging.error(line)
+        else:
+            # We always want to know what gets logged to stderr
+            logging.debug(line)
+
+
+def realtimeCommand(
```

Comment:
Extract this functionality into a separate module and use it for both cli and node